### ISSUE TITLE
Streaming interface

### ISSUE THREAD

**Issue by nickkaltner** [dashbitco/table#20]
This is just a suggestion more than anything.  I am wondering if it makes sense to have a Table.Reader.Streaming or similar where you have the implementation of the protocol do a next_page(data) until it returns nil.

My use case is mapping it to elasticsearch results that require a scroll to be opened for a large result set.  This would mean the consumer could decide if it should collect all results straight away or if it can deal with just the first page.

We could make the producer execute the whole scroll up front but i'm not sure that would always make sense.

**Comment by josevalim** [comment]
I believe you already do that today. The table reader protocol allows you to return enumerables, which you can consume lazily.

**Comment by nickkaltner** [comment]
Thanks José - I suspect my understanding is just incomplete.  

I looked at the Enumerable protocol, https://hexdocs.pm/elixir/Enumerable.html and it seems that I need to know the count of the pages or items returned eg. it's finite and known.  

So I looked into Streams (I think they do this because they are lazy) and the dializer spec says they return enumerables.  So i'm a little confused. Will look into it further! thanks for the tip
